test/lib/ansible_test/_internal/ansible_util.py

```diff
@@ -85,6 +85,10 @@ def ansible_environment(args, color=True, ansible_config=None):
         # the correct python interpreter is already selected using the sys.executable used to invoke ansible
         ansible.update(dict(
             ANSIBLE_CONNECTION_PATH=os.path.join(ANSIBLE_TEST_DATA_ROOT, 'injector', 'ansible-connection'),
+            # give TQM worker processes time to report code coverage results
+            # without this the last task in a play may write no coverage file, an empty file, or an incomplete file
+            ANSIBLE_WORKER_SHUTDOWN_POLL_COUNT='10',
+            ANSIBLE_WORKER_SHUTDOWN_POLL_DELAY='1',
```

Comment:
In my quick testing, 1 second is wayyyyyy more than we need. I set the poll count to 2 (so we sleep once, see comment above), and can go down to `0.02` without triggering the warning. I was going to suggest `20` and `0.5`, but now I'm thinking something like `100` and `0.1`. Same maximum time as what you had, but trims 0.9s of laggy-ness.

Reply:
I agree this makes more sense for reducing lag, although it may slightly increase CPU usage due to the additional polling. Thanks for verifying it works with a lower sleep.

I'm also thinking that this should always be enabled for testing, not just for coverage. If we're seeing the warning then that's an indication that a worker has hung after reporting results, which is something worth knowing.